src/main/java/dev/joopie/jambot/api/spotify/ApiSpotifyService.java

```diff
@@ -47,7 +57,8 @@ public void initSpotifyAccessToken() {
         }
     }
 
-    public Optional<String> getTrack(String link) {
+    public Optional<Track> getTrack(String link) {
+        Optional <Track> track = Optional.empty();
```

Comment:
I can't use final here because this variable is getting set in this method. Also, final var makes me to return an Optional<Object> which causes for problems down the line where I need to cast this into a Track object. I will check the other occurrences in the code and fix those if possible.

Reply:
Edit; removed the local variable now. Not needed with an instant return